### PR TITLE
sql: add switch for check constraint | tidb-test=pr/2160

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #41711 

Problem Summary:

### What is changed and how it works?
Add switch for check constraint, the default setting is `OFF`
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add switch for check constraint.
```

### CODE REVIEW

**Review by CbcWestwolf — APPROVED** [review]

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -2751,6 +2751,12 @@ var defaultSysVars = []*SysVar{
 			return nil
 		},
 	},
+	{Scope: ScopeGlobal, Name: TiDBEnableCheckConstraint, Value: BoolToOnOff(DefTiDBEnableCheckConstraint), Type: TypeBool, SetGlobal: func(ctx context.Context, vars *SessionVars, s string) error {
```
Comment (fzzf678) [inline sessionctx/variable/sysvar.go:2754]: @CbcWestwolf PTAL at if this variable is defined and used correctly

Comment (CbcWestwolf) [reply sessionctx/variable/sysvar.go:2754]: Yes, it is the way we use it.

File: executor/set_test.go [hunk 2]
```diff
@@ -2124,3 +2124,17 @@ func TestSetMppExchangeCompressionModeVariable(t *testing.T) {
 		require.Equal(t, warnings[0].Err.Error(), "mpp exchange compression won't work under current mpp version 0")
 	}
 }
+
+func TestSetEnableCheckConstraint(t *testing.T) {
```
Comment (CbcWestwolf) [inline executor/set_test.go:2128]: This test function is unnecessary since our system variable framework promises this cast.

**Review by CbcWestwolf — APPROVED** [review]

**Review by TonsnakeLin — APPROVED** [review]
LGTM

File: executor/set_test.go (hunk 2 above)
Comment (fzzf678) [reply executor/set_test.go]: updated

**Pull request merged**